### PR TITLE
[Breaking] new calcite-link component - breaking change for calcite-button

### PR DESCRIPTION
**Pull request by macandcheese**
https://github.com/Esri/calcite-components/issues/388

- **BREAKING** Removes appearance="inline" from calcite-button
- Cleans up calcite-button
- Creates new calcite-link component
- Link now inherits size and font-weight from container
- Because calcite-link is rendered as an inline item, it does not have a "scale" prop
- Adds test, storybook, dev demo
- Updates demos and components to use new component

Also fixes some line height issues the old appearance="inline" on calcite-button experienced. I know there had been some suggestions from Arjav around changing the styling for accessibility reasons for links.... this does not do that yet, but we should address that at some point...

### CODE REVIEW

**Comment by macandcheese** [comment]
Hmmm, Stencil must have changed the way they document prop values, looks like "/" is now "or" in all generated readme's, this must be the first PR to have built a Storybook and then submitted after version bump. I can clean those out of here if needed...

File: src/components/calcite-link/readme.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# calcite-button
```
Comment (paulcpederson) [inline src/components/calcite-link/readme.md]: this should probably say calcite-link, though

Comment (macandcheese) [reply src/components/calcite-link/readme.md]: D'oh

**Comment by macandcheese** [comment]
@Esri/calcite-components any other thoughts here?

**Comment by julio8a** [comment]
Let's merge this one in.

**Comment by macandcheese** [comment]
It's a tiny bit wonky with the new focus state and icons when combined, but I'd be in favor of getting it in and then updating style afterwards, since if folks use a `calcite-button` with `appearance=inline` this will be a breaking change for them, and I'd prefer to at least get the component in and update styles afterwards.

**Comment by paulcpederson** [comment]
@macandcheese do you want to just axe the `xs` and `xl` scale sizes so we don't have to remove them later? Feel free to merge this in, though, IMO

**Comment by macandcheese** [comment]
Cool, just fixed merge conflicts. Will save nixing `xs` and `xl` from button for another update, since we'll tweak the spacing / font size to use the new values at that point. `calcite-link` doesn't actually accept a scale prop because it's rendered inline and inherits from container / surrounding text.

**Pull request merged**